pkg/providers/gcp/provider_postgres.go

```diff
@@ -3,63 +3,229 @@ package gcp
 import (
 	"context"
 	"fmt"
+	"time"
+
 	"github.com/integr8ly/cloud-resource-operator/apis/integreatly/v1alpha1"
-	"github.com/integr8ly/cloud-resource-operator/apis/integreatly/v1alpha1/types"
+	croType "github.com/integr8ly/cloud-resource-operator/apis/integreatly/v1alpha1/types"
+	"github.com/integr8ly/cloud-resource-operator/pkg/annotations"
 	"github.com/integr8ly/cloud-resource-operator/pkg/providers"
 	"github.com/integr8ly/cloud-resource-operator/pkg/resources"
+	errorUtil "github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/api/option"
+	sqladmin "google.golang.org/api/sqladmin/v1beta4"
+	v1 "k8s.io/api/core/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	"time"
 )
 
-const postgresProviderName = "gcp-cloudsql"
+var _ providers.PostgresProvider = (*PostgresProvider)(nil)
+
+const (
+	postgresProviderName         = "gcp-cloudsql"
+	projectID                    = "rhoam-317914"
```

Comment:
On this particular point I agree, but I think it might just be easiest that I quickly adjust this when I rebase on top of these changes - I can refactor to get the project from the infrastructure (https://github.com/adam-cattermole/cloud-resource-operator/blob/MGDAPI-4734/pkg/resources/cluster.go#L42-L51) until we implement the strategy.

Reply:
I have created a jira ticket to complete this work when we are ready. https://issues.redhat.com/browse/MGDAPI-4843